hedera-node/src/main/java/com/hedera/services/store/contracts/precompile/DirectCallsInfoProvider.java

```diff
@@ -0,0 +1,82 @@
+package com.hedera.services.store.contracts.precompile;
+
+/*-
+ * ‌
+ * Hedera Services Node
+ * ​
+ * Copyright (C) 2018 - 2022 Hedera Hashgraph, LLC
+ * ​
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * ‍
+ */
+
+import org.apache.tuweni.bytes.Bytes;
+import org.hyperledger.besu.datatypes.Address;
+import org.hyperledger.besu.datatypes.Wei;
+import org.hyperledger.besu.evm.frame.MessageFrame;
+import org.hyperledger.besu.evm.log.Log;
+
+import java.util.List;
+
+public record DirectCallsInfoProvider(PrecompileMessage precompileMessage) implements InfoProvider {
+	@Override
+	public Wei getValue() {
+		return precompileMessage.getValue();
+	}
+
+	@Override
+	public MessageFrame messageFrame() {
+		return null;
```

Comment:
To avoid returning null directly, wouldn't it be better if we have `Optional<MessageFrame>`?